### PR TITLE
Extract Value to separate file

### PR DESCRIPTION
**Pull request by charleso**

### CODE REVIEW

File: ivory-core/src/main/scala/com/ambiata/ivory/core/Value.scala [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+      case (vp: PrimitiveValue, EncodingPrim(ep)) => validateEncodingPrim(vp, ep)
+      case (vp: PrimitiveValue, _)              => fail
+      case (StructValue(vs), EncodingStruct(es)) => validateStruct(vs, es)
+      case (StructValue(_),  _)                 => fail
+      case (ListValue(v), EncodingList(l))      => v.foldMap(validateEncodingSub(_, l.encoding))
+      case (ListValue(_),    _)                 => fail
+    }
+  }
+
+  def toThrift(value: Value): ThriftFactValue =
```
Comment (charleso) [inline ivory-core/src/main/scala/com/ambiata/ivory/core/Value.scala]: Not sure if I should create a sub `value` package will separate classes for these as well?

Comment (charleso) [inline ivory-core/src/main/scala/com/ambiata/ivory/core/Value.scala]: > Not sure if I should create a sub value package will separate classes for these as well?

Let me rephrase that. I'd like to do something to keep this file from ballooning out (eg. json). Is a sub-package the preferred way?

**Comment by charleso** [comment]
@markhibberd Ping.

**Comment by markhibberd** [comment]
Change looks good, but I am strongly adverse to sub-packages.
